Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+                    // TODO Update this after KHR_materials_pbrSpecularGlossiness extension is supported
+                    //if (gltfObject.extensionsUsed[i].Equals("KHR_materials_pbrSpecularGlossiness"))
+                    //{
+                    //    for (int j = 0; j < gltfObject.materials.Length; j++)
+                    //    {
+                    //        if (!string.IsNullOrEmpty(gltfObject.materials[i].name) &&
+                    //            gltfObject.materials[i].name == extensionUsed.Key)
+                    //        {
+                    //            gltfObject.materials[i].Extensions.Add(gltfObject.extensionsUsed[i], extensionUsed.Value);
+                    //            var extension = JsonUtility.FromJson<KHR_Materials_PbrSpecularGlossiness>(extensionUsed.Value);
+                    //            extension.ElementName = gltfObject.materials[i].name;
+                    //            gltfObject.RegisteredExtensions.Add(extension);
+                    //        }
+                    //    }
+                    //}
+                    //else
+                    {
+                        Debug.LogWarning($"Unsupported Extension: {gltfObject.extensionsUsed[i]}");
+                    }
+                }
+            }
+
+            var meshPrimitiveAttributes = GetGltfMeshPrimitiveAttributes(jsonString);
+
+            for (int i = 0; i < gltfObject.meshes.Length; i++)
+            {
+                for (int j = 0; j < gltfObject.meshes[i].primitives.Length; j++)
+                {
+                    gltfObject.meshes[i].primitives[j].Attributes = JsonUtility.FromJson<GltfMeshPrimitiveAttributes>(meshPrimitiveAttributes[0]);
+                    meshPrimitiveAttributes.Remove(meshPrimitiveAttributes[0]);
+                }
+            }
+
+            return gltfObject;
+        }
+
+        /// <summary>
+        /// Gets a glTF object from the provided path
+        /// </summary>
+        /// <param name="glbData">Raw glb byte data.</param>
+        /// <returns><see cref="GltfObject"/></returns>
+        /// <remarks>Returned <see cref="GltfObject"/> still needs to be initialized using <see cref="ConstructGltf.ConstructAsync"/>.</remarks>
+        public static GltfObject GetGltfObjectFromGlb(byte[] glbData)
+        {
+            const int stride = sizeof(uint);
+
+            var magicNumber = BitConverter.ToUInt32(glbData, 0);
+            var version = BitConverter.ToUInt32(glbData, stride);
+            //var length = BitConverter.ToUInt32(glbData, stride * 2);
+
+            if (magicNumber != GltfMagicNumber)
+            {
+                Debug.LogError("File is not a glb object!");
+                return null;
+            }
+
+            if (version != 2)
+            {
+                Debug.LogError("Glb file version mismatch! Glb must use version 2");
+                return null;
+            }
+
+            var chunk0Length = BitConverter.ToUInt32(glbData, stride * 3);
```

Comment:
its probably worth calling out what chunk0 and chunk1 contain here as well as what influences their sizes in a comment.

Reply:
It's part of the glTF standard. It's how they defined it in the glb data.

Reply:
After looking at this longer, it seems like a developer can infer what's contained in the glbData by looking further down in the parsing steps. If this header, chunk, chunk scheme was more complicated it may be worth calling out what all the different components contain, but it seems reasonable that this amount of detail isn't needed.